### PR TITLE
Connection to database. (V1-005)

### PR DESCRIPTION
**Pull request by wooz1ewu**
# Overview 
> 1. Implemented connection middleware with error passing.  
> 1. Configured migration scripts.  
> 1. Configured initial migration. (will be updated soon with db collections structure updates)  
> 1. Configured User model.  
> 1. Added all the neccessary types and interfaces.  
> 1. Added env varuables (connection strings).  
> 1. Added constants for connection&hashing configuration.

### CODE REVIEW

File: backend/migrations/20211219190858-init_collections.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const bcrypt = require('bcrypt');
+
+const saltRounds = 10;
```
Comment (k-krasilnikova) [inline backend/migrations/20211219190858-init_collections.js:3]: uppercase please!

File: backend/migrations/20211219190858-init_collections.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    // await db.collection('courses').drop();
+    // await db.collection('userSkills').drop();
+    // await db.collection('userCourses').drop();
+  },
+};
+
+const DEFAULT_USERS_DOCS = [
+  {
+    name: 'admin',
+    password: 'admin',
```
Comment (k-krasilnikova) [inline backend/migrations/20211219190858-init_collections.js:30]: We can't store password as it is.
Hash please this password and store it in `passwordHash` field.

File: backend/src/interfaces/db/entities.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { ObjectId } from 'mongoose';
+
+import { USER_ROLES } from 'config/constants';
+
+interface User {
+  _id?: ObjectId;
+  username: string;
+  password: string;
```
Comment (k-krasilnikova) [inline backend/src/interfaces/db/entities.d.ts:8]: passwordHash!

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

File: backend/migrations/20211219190858-init_collections.js (hunk 2 above)
Comment (wooz1ewu) [reply backend/migrations/20211219190858-init_collections.js:30]: It's being hashed during proccess of migrating up. Or you want password to be stored as hash right in `DEFAULT_USER_DOCS`?

File: backend/src/db/models/User.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  fullName: { type: String, required: true },
+  skills: [{ type: Schema.Types.ObjectId, ref: 'UserSkill' }],
+  courses: [{ type: Schema.Types.ObjectId, ref: 'UserCourse' }],
+  employees: [{ type: Schema.Types.ObjectId, ref: 'User' }],
+  avatar: { data: Buffer, contentType: String, required: false },
+  birthday: { type: Date, required: true },
+  skype: { type: String, required: true, default: 'not provided' },
+});
+
+const UserModel = model<User>('User', schema);
```
Comment (SaXaPhonist) [inline backend/src/db/models/User.ts:21]: you do migration into "users" collection not "User"

File: backend/src/db/models/User.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { Schema, model } from 'mongoose';
+
+import { User } from 'interfaces/db/entities';
+
+const schema = new Schema<User>({
+  _id: { type: Schema.Types.ObjectId, required: true },
+  username: { type: String, required: true },
```
Comment (SaXaPhonist) [inline backend/src/db/models/User.ts:7]: do we really need "username" field maybe replace it with "email" field

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/db/models/User.ts (hunk 4 above)
Comment (wooz1ewu) [reply backend/src/db/models/User.ts:21]: That is actually the name of model, not collection, as far as I know 🤔

File: backend/migrations/20211219190858-init_collections.js (hunk 2 above)
Comment (k-krasilnikova) [reply backend/migrations/20211219190858-init_collections.js:30]: Oh, you're right!
Let's leave it in such way

**Review by SaXaPhonist — DISMISSED** [review]

**Review by zhuk-vezunchikz — DISMISSED** [review]

**Review by k-krasilnikova — APPROVED** [review]

**Review by SaXaPhonist — APPROVED** [review]

**Pull request merged**